index.js

```diff
@@ -125,19 +122,14 @@ function escapeRegExp(s) {
     return s.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&');
 }
 
-function matchFromRegex(s, options) {
-  if (options.regExpOptions) {
-    return new RegExp(s.source, options.regExpOptions)
-  } else {
-    var flags = s.toString().replace(/\/[^\/].*\//, '')
-    // If there is no global flag then there can only be one match
-    if (flags.indexOf('g') < 0) {
-      options.limit = 1;
-    }
-    return new RegExp(s.source, flags)
+function matchFromRegex(regex, options) {
+  // If there is no global flag then there can only be one match
+  if (!regex.global) {
```

Comment:
oh nice. my bad. looks good then. for some reason i thought this said options.global. Either way i didnt know about that property :)